### ISSUE TITLE
ENS protocol redirect server

### ISSUE THREAD

**Issue by mikehelmick** [google/exposure-notifications-verification-server#549]
### TL;DR
A prototype for this exists in PR #546 

The current [ens protocol spec](https://github.com/google/exposure-notifications-verification-server/blob/main/docs/ens-spec.md) is not clickable in all messaging apps that may be used to receive SMS deep links on mobile devices.

In order to ensure that all users have a clickable link, we will provide a redirect server that verification server operators can use to 

### Design

**Proposal**

Simple URL redirect service. This needs to support custom subdomain mapping to ENS link.

Having unique subdomains allows per-region applications to to register that intent and handle it in the mobile OS.

As an example.

https://us-wa.example.com/v?c=1234abcd5678efgh

will redirect to 

ens://v?c=1234abcd5678efgh&r=US-WA

Configuration will consist of 2 elements
URI prefix to simply rewrite to ens://
and the region to append
will add "&r=REGION"

redirect requests that reach the server and do not match will return a 404.

**Alternatives considered**

the existing ens:// schema has proven not to work. 

**Resources**

[ens protocol spec](https://github.com/google/exposure-notifications-verification-server/blob/main/docs/ens-spec.md)

**Additional information**

For production deployment, request logging should be disabled.

/kind enhancement
/assign

**Comment by mikehelmick** [comment]
PR for server itself is #546 

Still to do

- Terraform
- Update main server w/ instructions for deep links

**Comment by mikehelmick** [comment]
/assign @icco 
is going to take on the terraform piece

**Comment by sethvargo** [comment]
@icco are there remaining items here that aren't tracked in other issues?

**Comment by icco** [comment]
Nope we should be good here. https://github.com/google/exposure-notifications-verification-server/pull/613 and https://github.com/google/exposure-notifications-verification-server/pull/615 we're the last fixes.

/close
